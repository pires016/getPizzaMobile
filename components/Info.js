import React from 'react';
import { View, Text } from 'react-native';


const Info = ({valor, styles}) => {
    return(
        <View>
            <Text style={styles}>Informações - {valor ? valor : 'Rita Lee'}</Text>
        </View>
    );
}

export default Info;