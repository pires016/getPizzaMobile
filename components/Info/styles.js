import { StyleSheet } from 'react-native';


const Info = ({valor, styles}) => {
    return(
        <View>
            <Text style={styles}>Informações - {valor ? valor : 'Up Up'}</Text>
        </View>
    );
}

export default Info;