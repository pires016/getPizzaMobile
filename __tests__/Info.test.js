import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Info from '../components/Info';

test('renders correctly Info component', () => {
    const tree = renderer.create(<Info />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('check value', () => {
    const value = "Raul";
    const tree = renderer.create(<Info valor={value} />);
    const instance = tree.root;
    expect(instance.findAllByType(Text)[0].props.children).toEqual(["Informações - ", value]);
});

test('check style - findAllByType', () => {
    const style = { color: "blue" };
    const tree = renderer.create(<Info styles={style} />);
    const instance = tree.root;
    expect(instance.findAllByType(Text)[0].props.style).toEqual(style);
});

test('check style - findByType', () => {
    const style = { color: "blue" };
    const tree = renderer.create(<Info styles={style} />);
    const instance = tree.root;
    expect(instance.findByType(Text).props.style).toEqual(style);
});
