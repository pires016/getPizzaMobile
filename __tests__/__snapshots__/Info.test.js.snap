// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Info component 1`] = `
<View>
  <Text>
    Informações - 
    Rita Lee
  </Text>
</View>
`;
